device/tt_silicon_driver.cpp

```diff
@@ -3025,11 +3098,40 @@ int tt_SiliconDevice::iatu_configure_peer_region (int logical_device_id, uint32_
     struct PCIdevice* pci_device = get_pci_device(logical_device_id);
     auto architecture_implementation = pci_device->hdev->get_architecture_implementation();
 
-    bar_write32(logical_device_id, architecture_implementation->get_arc_csm_mailbox_offset() + 0 * 4, region_id_to_use);
-    bar_write32(logical_device_id, architecture_implementation->get_arc_csm_mailbox_offset() + 1 * 4, dest_bar_lo);
-    bar_write32(logical_device_id, architecture_implementation->get_arc_csm_mailbox_offset() + 2 * 4, dest_bar_hi);
-    bar_write32(logical_device_id, architecture_implementation->get_arc_csm_mailbox_offset() + 3 * 4, region_size);
-    arc_msg(logical_device_id, 0xaa00 | architecture_implementation->get_arc_message_setup_iatu_for_peer_to_peer(), true, 0, 0);
+    // BR: ARC doesn't work yet on Blackhole, so programming ATU directly. Should be removed when arc starts working.
+    if (arch_name == tt::ARCH::BLACKHOLE) {
+        uint64_t base_addr = region_id_to_use*region_size;
+        uint64_t base_size = (region_id_to_use+1)*region_size;
+        uint64_t limit_address = base_addr + base_size - 1;
+
+        uint32_t region_ctrl_1 = 1 << 13; // INCREASE_REGION_SIZE = 1
+        uint32_t region_ctrl_2 = 1 << 31; // REGION_EN = 1
+        uint32_t region_ctrl_3 = 0;
+        uint32_t base_addr_lo = base_addr & 0xffffffff;
+        uint32_t base_addr_hi = (base_addr >> 32) & 0xffffffff;
+        uint32_t limit_address_lo = limit_address & 0xffffffff;
+        uint32_t limit_address_hi = (limit_address >> 32) & 0xffffffff;
+
+        uint64_t iatu_index = 0;
+        uint64_t iatu_base = UNROLL_ATU_OFFSET_BAR + iatu_index * 0x200;
```

Comment:
What is 0x200 constant for?

Reply:
I don't want to leave a comment/constant guessing it. But from the code, I'd say that there are different registers for different ATU regions (i.e. you can setup multiple address range translations), and they are spaced out by this amount. This was copied from example python code.